### PR TITLE
utils_net: simple ipv4 address generator

### PR DESCRIPTION
**Pull request by balamuruhans**
API to generate ipv4 address and can be used inside tests for
assigning static IP address to guests/hosts

Signed-off-by: Balamuruhan S <bala24@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by balamuruhans** [comment]
@luckyh done.

**Comment by luckyh** [comment]
@balamuruhans I found that package `netaddr` have provided an implementation for this, and the package has already been imported by `utils_net.py`. So sorry, I think you'd better change the code, and make it as a wrapper.

References:
https://netaddr.readthedocs.io/en/latest/api.html#ip-functions-and-generators
https://netaddr.readthedocs.io/en/latest/api.html#ip-networks-and-subnets

**Comment by balamuruhans** [comment]
@luckyh, Thanks for the suggestion, made the changes.

**Comment by CongLi** [comment]
Acked-by: Cong Li <coli@redhat.com>

**Comment by luckyh** [comment]
@balamuruhans why not make it for common use, like:
```python
def gen_ipv4_addr(addr, exclude=[]):
    net = netaddr.IPNetwork(addr)
    for ip in net:
        if str(ip) in exclude:
            continue
        yield (str(ip),
               str(net.netmask),
               str(net.broadcast))

>>> NETADDR = "10.0.0.0/24"
>>> EXCLUDE = ["10.0.0.1", "10.0.0.255"]
>>> g = gen_ipv4_addr(NETADDR, EXCLUDE)
>>> for i, m, b in g:
...     print i, m, b
...
10.0.0.0 255.255.255.0 10.0.0.255
10.0.0.2 255.255.255.0 10.0.0.255
...
```

**Comment by balamuruhans** [comment]
@luckyh, Thanks for the idea, but the API `gen_ipv4_addr()` should generate ip address similar to a API generates mac address in `utils_net` and I feel returning netmask and broadcast is not the right way.

**Comment by luckyh** [comment]
@balamuruhans ok, we can only return `ip`.

The main point is I think user must specify ip range or give a network definition to calculate the range, otherwise, the code have to decide a range by hardcoding, which is not a good idea.

File: virttest/utils_net.py [hunk 1]
```diff
@@ -3080,6 +3080,30 @@ def generate_mac_address_simple():
...
+    """
+    ip_regex = "^\d+.\d+.\d+.\d+$"
+    if not re.match(ip_regex, network_num):
+        network_num = "10.0.0.0"
+    if not exclude_ips:
+        exclude_ips.append(network_num)
+        exclude_ips.append('.'.join(network_num.split('.')[0:3]) + ".%s" %
+                           str(1))
+        exclude_ips.append(('.'.join(network_num.split('.')[0:3]) + ".%s" %
+                           str(255)))
```
Comment (luckyh) [inline virttest/utils_net.py]: The above two statements are not always correct for every subnet.

Comment (balamuruhans) [inline virttest/utils_net.py]: this is for default value when `network_prefix = "24"`, if user gives different network prefix for his required subnet, then user can also give `exclude_ips` too.

Comment (luckyh) [inline virttest/utils_net.py]: If so, I suggest appending the judgement to L3095. And it is better to make `exclude_ips` as a `set()`.

Comment (luckyh) [inline virttest/utils_net.py]: If so, I suggest appending the judgement to L3095. And it is better to make `exclude_ips` as a `set()`.

Comment (balamuruhans) [inline virttest/utils_net.py]: @luckyh Agreed, I have made the changes as suggested.

**Comment by balamuruhans** [comment]
Thank you @luckyh

**Pull request merged**